lib/binary/serializer.cpp

```diff
@@ -24,22 +27,32 @@ void serializer::emit(std::byte b) {
 
 void serializer::emit_string(std::string const &s) {
   if (intern_table_.find(s) == intern_table_.end()) {
-    emit(direct_string_prefix_);
-
-    intern_table_[s] = next_idx_;
+    emit_direct_string(s);
+  } else {
+    int64_t previous = intern_table_.at(s);
+    int64_t diff = (next_idx_ + 5) - previous;
```

Comment:
I think I am missing something. Could you explain the reason for changing `next_idx_ + 4` to `next_idx_ + 5`?

Reply:
The reason is that we want to compute the diff _before_ the prefix (`0x01` or `0x02`) has been emitted to the buffer, so we're a byte further back in the buffer.

Reply:
I've added a comment to this code explaining where the 5-byte offset comes from.

Reply:
That sounds correct, thanks for explaining!